aiidalab_widgets_base/wizard.py

```diff
@@ -0,0 +1,212 @@
+        self.header = ipw.HBox(
+            children=[self.back_button, self.reset_button, self.next_button]
+        )
+
+        super().__init__(children=[self.header, self.accordion], **kwargs)
+
+    def _update_titles(self):
+        for i, (title, widget) in enumerate(zip(self.titles, self.accordion.children)):
+            icon = self.icons().get(widget.state, str(widget.state).upper())
+            self.accordion.set_title(i, f"{icon} Step {i+1}: {title}")
+
+    def _consider_switch(self, _=None):
+        with self.hold_trait_notifications():
+            index = self.accordion.selected_index
+            last_step_selected = index + 1 == len(self.accordion.children)
+            selected_widget = self.accordion.children[index]
+            if (
+                selected_widget.auto_next
+                and not last_step_selected
+                and selected_widget.state == WizardAppWidgetStep.State.SUCCESS
+            ):
+                self.accordion.selected_index += 1
+
+    def _update_step_state(self, _):
+        with self.hold_trait_notifications():
+            self._update_titles()
+            self._update_buttons()
+            self._consider_switch()
+
+    @traitlets.observe("selected_index")
+    def _observe_selected_index(self, change):
+        "Activate/deactivate the next-button based on which step is selected."
+        self._update_buttons()
+
+    def can_reset(self):
+        steps = [
+            self.accordion.children[idx] for idx in range(len(self.accordion.children))
+        ]
+
+        if any(not step.can_reset() for step in steps):
+            return False
+
+        if any(step.state is not WizardAppWidgetStep.State.INIT for step in steps):
+            return True
+
+    def _update_buttons(self):
+        with self.hold_trait_notifications():
+            index = self.accordion.selected_index
+            if index is None:
+                self.back_button.disabled = True
+                self.next_button.disabled = True
+                self.reset_button.disabled = True
+            else:
+                first_step_selected = index == 0
+                last_step_selected = index + 1 == len(self.accordion.children)
+                selected_widget = self.accordion.children[index]
+
+                self.back_button.disabled = (
+                    first_step_selected
+                    or selected_widget.state
+                    in (
+                        WizardAppWidgetStep.State.ACTIVE,
+                        WizardAppWidgetStep.State.SUCCESS,
+                        WizardAppWidgetStep.State.FAIL,
+                    )
+                )
+                self.next_button.disabled = (
+                    last_step_selected
+                    or selected_widget.state is not WizardAppWidgetStep.State.SUCCESS
+                )
+
+                self.reset_button.disabled = not self.can_reset()
+
+    def reset(self, step=0):
+        """Reset the app down to the given step.
```

Comment:
Point 1: Good catch, I will fix the doc-string.

Point 2: You mean start from the last step and up? Otherwise we would end up in an inconsistent state. In principle, I agree, however we use `hold_sync()` to actually prevent any synchronization between steps during the reset, so we generate a given state and then only after the hold is lifted we perform the consistency checks.

Reply:
> You mean start from the last step and up?

Yes.

>  In principle, I agree, however we use hold_sync() to actually prevent any synchronization between steps during the reset, so we generate a given state and then only after the hold is lifted we perform the consistency checks.


Ok, true. Then it is up to you if you want to reverse the order.